web/src/components/EpisodeViewer.vue

```diff
@@ -250,6 +256,34 @@ const checkProgress = async (force?: boolean) => {
     }
 }
 
+let playedOverHalfCounter = 0
```

Comment:
I don't understand why and how this is used

Reply:
This is essentially used to prevent spamming the endpoint.

Each tick, every ~10th of a second or so, the `onProgress` hook is called by the video player, and I check if the video progress is > 50%.

Each tick `playedOverHalfCounter` is incremented, and when the counter is exactly 10, I call the `/Statistics/watched` endpoint.
This makes sure we don't spam the endpoint, and the user has to keep playing for a second or so after rounding 50% for the endpoint to be called.

Overengineered?